src/systems/systems/src/electrical/battery.rs

```diff
@@ -0,0 +1,180 @@
+use super::{
+    ElectricalStateWriter, Potential, PotentialSource, PotentialTarget, ProvideCurrent,
+    ProvidePotential,
+};
+use crate::simulation::{SimulationElement, SimulatorWriter};
+use uom::si::{
+    electric_charge::ampere_hour, electric_current::ampere, electric_potential::volt, f64::*,
+};
+
+pub struct Battery {
+    writer: ElectricalStateWriter,
+    number: usize,
+    input: Potential,
+    charge: ElectricCharge,
+}
+impl Battery {
+    const MAX_ELECTRIC_CHARGE_AMPERE_HOURS: f64 = 23.0;
+
+    pub fn full(number: usize) -> Battery {
+        Battery::new(
+            number,
+            ElectricCharge::new::<ampere_hour>(Battery::MAX_ELECTRIC_CHARGE_AMPERE_HOURS),
+        )
+    }
+
+    pub fn empty(number: usize) -> Battery {
+        Battery::new(number, ElectricCharge::new::<ampere_hour>(0.))
+    }
+
+    fn new(number: usize, charge: ElectricCharge) -> Self {
+        Self {
+            writer: ElectricalStateWriter::new(&format!("BAT_{}", number)),
+            number,
+            input: Potential::None,
+            charge,
+        }
+    }
+
+    pub fn is_full(&self) -> bool {
+        self.charge >= ElectricCharge::new::<ampere_hour>(Battery::MAX_ELECTRIC_CHARGE_AMPERE_HOURS)
+    }
+
+    pub fn input_potential(&self) -> Potential {
+        self.input
+    }
+}
+potential_target!(Battery);
+impl PotentialSource for Battery {
+    fn output_potential(&self) -> Potential {
+        if self.input.is_unpowered() && self.charge > ElectricCharge::new::<ampere_hour>(0.) {
+            Potential::Battery(self.number)
+        } else {
+            Potential::None
+        }
+    }
+}
+impl ProvideCurrent for Battery {
+    fn current(&self) -> ElectricCurrent {
+        // TODO: Replace with actual values once calculated.
+        ElectricCurrent::new::<ampere>(0.)
+    }
+
+    fn current_normal(&self) -> bool {
+        // TODO: Replace with actual values once calculated.
+        self.output_potential().is_powered()
+    }
+}
+impl ProvidePotential for Battery {
+    fn potential(&self) -> ElectricPotential {
+        // TODO: Replace with actual values once calculated.
```

Comment:
Can't wait for this! Seeing the nuances there affect numbers in the cockpit is going to be amazing.